iroh-p2p/src/node.rs

```diff
@@ -1103,7 +1122,13 @@ mod tests {
                 }
             };
             let mut network_config = Config::default_with_rpc(rpc_client_addr.clone());
-            network_config.libp2p.listening_multiaddr = self.addr;
+
+            if let Some(addr) = self.addr {
+                network_config.libp2p.listening_multiaddr = addr;
+            } else {
+                network_config.libp2p.listening_multiaddr = "/ip4/0.0.0.0/tcp/0".parse().unwrap();
```

Comment:
Before you had some listening on 127.0.0.1 instead of all interfaces.  You could still force having to give an addr to `::new()` but describe that you can use `0` as port` and how to retrieve the final addr.  That way those tests are not needlessly exposing themselves entirely to the internet (or local coffee shop wifi where we happen to be running some tests ;))

Reply:
good point!

Reply:
Although I do like the API of not having to pass in an address, but rather have the default address be `/ip4/127.0.0.1/tcp/0` instead of `/ip4/0.0.0.0/tcp/0`